cwl_wes/custom_config.py

```diff
@@ -29,32 +114,140 @@ class WorkflowTypeVersionConfig(FOCABaseConfig):
     Args:
         workflow_type_version: List of one or more acceptable versions for the
             workflow type.
+        
+    Attributes:
+        workflow_type_version: List of one or more acceptable versions for the
+            workflow type.    
+
+    Raises:
+        pydantic.ValidationError: The class was instantianted with an illegal
+            data type.
+
+    Example:
+        >>> WorkflowTypeVersionConfig(
+        ...     workflow_type_version=['v1.0']
+        ... )
+        WorkflowTypeVersionConfig(workflow_type_version=['v1.0'])
     """
     workflow_type_version: Optional[List[str]] = []
 
 
 class DefaultWorkflowEngineParameterConfig(FOCABaseConfig):
     """Model for default workflow engine parameters.
+
     Args:
         name: Parameter name.
         type: Parameter type.
         default_value: Stringified version of default parameter.
+
     Attributes:
         name: Parameter name.
         type: Parameter type.
         default_value: Stringified version of default parameter.
+
+    Raises:
+        pydantic.ValidationError: The class was instantianted with an illegal
+            data type.
+
+    Example:
+        >>> DefaultWorkflowEngineParameterConfig(
+        ...     name='name',
+        ...     type='str',
+        ...     default_value='default'
+        ... )
+        DefaultWorkflowEngineParameterConfig(name='name', type='str', default_value='default')
     """
     name: Optional[str]
     type: Optional[str]
     default_value: Optional[str]
 
 
 class TagsConfig(FOCABaseConfig):
+    """Model for service info tag configuration.
+
+    Args:
+        known_tes_endpoints: Valid TES endpoints.
+        app_version: Application version.
+
+    Attributes:
+        known_tes_endpoints: Valid TES endpoints.
+        app_version: Application version.
+
+    Raises:
+        pydantic.ValidationError: The class was instantianted with an illegal
+            data type.
+
+    Example:
+        >>> TagsConfig(
+        ...     known_tes_endpoints='https://tes.endpoint',
+        ...     app_version='0.1.x',
+        ... )
+        TagsConfig(known_tes_endpoints='https://tes.endpoint', app_version='0.1.x')
+    """
     known_tes_endpoints: str = 'https://tes.tsi.ebi.ac.uk/|https://tes-dev.tsi.ebi.ac.uk/|https://csc-tesk.c03.k8s-popup.csc.fi/|https://tesk.c01.k8s-popup.csc.fi/'
     app_version: str = '0.15.0'
```

Comment:
Not too happy with this, as we should source this from the app's version automatically (like we do in FOCA), not define in multiple places. It's nothing for the user/admin to configure anyway.